test/parallel_api/experimental/asynch_include_order_error.pass.cpp

```diff
@@ -0,0 +1,55 @@
+// -*- C++ -*-
+//===-- async.pass.cpp ----------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+// Including `async` after `execution` and `numeric` compiles successfully.
+#include <oneapi/dpl/async>
+#include <oneapi/dpl/execution>
+#include <oneapi/dpl/numeric>
```

Comment:
The implementation of the test does not match the comment, and also the order of includes is the same as in the other test.

Reply:
It was the reproducer of compile error from customer.
Removed.

Reply:
Just to note, I have not suggested that you should remove it. I only pointed out that there is a mismatch of the description (as I read it) and the behavior, and also that another test has exactly the same order of includes while we perhaps want to test different orders.

It's up to you to decide if a special regression test is needed or not. Actually, we have a number of header inclusion order tests in `test/general`, adding one more for `<async>` might be completely reasonable.

Reply:
I have removed this new test (with reproducer from the user) because I also changed the order of includes in the already existing test.
That is the reason why you see the same order of includes in the tests.

Two new tests were added too.

Reply:
But I should note that our approach in these tests of inclusion only partially generates and catching compile time error.

For example, by this new added tests on `main` branch I found only one error of two - when I have compiled this two new tests with `-DONEDPL_BACKEND=dpcpp`.

As in user example from https://github.com/oneapi-src/oneDPL/issues/881 the compile time error not on inclusion of `async` header but on later call of something from `async` header algorithms:
```
auto f = oneapi::dpl::experimental::reduce_async(policy, v, v + n, T(0), std::plus());
```

It's just for you info.

Reply:
Adding more checks, including API calls, to the body of the `header_*` tests is completely fine.

Reply:
Fixed: some small test functional inserted into
[test/general/header_inclusion_order_async_0.pass.cpp](https://github.com/oneapi-src/oneDPL/blob/26c01a7c9090e67c9cf0c74f5d83cbe01773149e/test/general/header_inclusion_order_async_0.pass.cpp)
[test/general/header_inclusion_order_async_1.pass.cpp](https://github.com/oneapi-src/oneDPL/blob/26c01a7c9090e67c9cf0c74f5d83cbe01773149e/test/general/header_inclusion_order_async_1.pass.cpp)